opentelemetry-api/src/opentelemetry/util/re.py

```diff
@@ -51,8 +52,8 @@ def parse_headers(s: str) -> Mapping[str, str]:
             continue
         # value may contain any number of `=`
         name, value = match.string.split("=", 1)
-        name = name.strip().lower()
-        value = value.strip()
+        name = unquote(name).strip().lower()
```

Comment:
There are use cases when the [key](https://github.com/w3c/baggage/blob/main/baggage/HTTP_HEADER_FORMAT.md#key) value is encoded?

Reply:
I wasn't sure about this either.
All of the `opentelemetry-{lang}` libraries that implement decoding seem to apply it to `key` as well as `value` (links to the relevant code are in the linked issue).